bot/app/database/db_objects.py

```diff
@@ -0,0 +1,214 @@
+    - what database item
+    - which operations incl. parameters
+- add logging, catch errors to log
+"""
+
+import sqlite3
+import logging
+import os
+
+logging.basicConfig(os.path.abspath('../log.txt', level=logging.DEBUG))
+
+DB_NAME = 'data.sqlite'
+
+class Event:
+
+    """
+    An instance of the Event relation 
+    """
+
+    def __init__(self, e_name, date, time, admi, desc, loca, e_pic_id):
+        
+        
+        self.e_name = e_name
+        self.date = date
+        self.time = time
+        self.admi = admi
+        self.desc = desc
+        self.loca = loca
+        self.e_pic_id = e_pic_id
+
+        self.insert_event()
+    
+    def insert_event(self):
+        
+        conn = sqlite3.connect(DB_NAME)
+        curs = conn.cursor()
+
+        try:
+            curs.execute("""
+                insert into Events values (?,?,?,?,?,?)""",
+                (self.e_name, self.date, self.time, self.admission, self.desc, \
+                self.e_pic_id))
+            self.conn.commit()
+        except sqlite3.IntegrityError as e:
+            logging.error(e)
+        
+        curs.close()
+        conn.close()
+        
+class Artist:
+
+    def __init__(self, a_name, webs, soundc, bandc, bio, a_pic_id):
+
+        self.a_name = a_name
+        self.webs = webs
+        self.soundc = soundc
+        self.bandc = bandc
+        self.bio = bio
+        self.a_pic_id = a_pic_id
+
+        self.insert_artist()
+        
+    def insert_artist(self):
+        
+        conn = sqlite3.connect(DB_NAME)
+        curs = conn.cursor()
+
+        try:
+            curs.execute("""
+                insert into Artists values (?,?,?,?,?,?)""",
+                (self.a_name, self.webs, self.soundc, self.bandc, self.bio, \
+                self.a_pic_id))
+            self.conn.commit() 
+        except sqlite3.IntegrityError as e:
+            logging.error(e)
+        
+        curs.close()
+        conn.close()
+
+class PlaysAt:
+
+    """
+    Instance represents Artist-Event connection (aka playsAt Table)
+    """
+
+    def __init__(self):
+        
+        self.a_name = input("Enter the Artist's name: ")
+        self.e_name = input("Enter the Event's name: ")
+        self.date = input("Enter the Event's date (format DD.MM.YYYY): ")
+
+        self.check_for_artist()
+        self.check_for_event()
+        self.insert_plays_at()
+
+    def check_for_artist(self):
+        
+        conn = sqlite3.connect(DB_NAME)
+        curs = conn.cursor()
+        
+        curs.execute("select * from Artists where a_name=?", (self.a_name, ))
+        artist = curs.fetchone()
+        
+        if artist:
+            print("Corresponding artist found! Check: \n", artist)
+            answ = input("Is this the right entry? [Y/n]" )
+            if answ == "Y":
+                pass
+            else:
+                self.a_name = input("Please re-enter artist-name: ")
+                check_for_artist()
```

Comment:
I would advise against calling this recursively:
- It will be stuck in an infinite loop in case the artist doesn't exist
- Closing the connection & cursor at the end could be problematic if somewhere down the call chain you get an exception.

Also I would not take user input inside a class function, but get the input with a wrapper function, similar to `new_event`  below. This gives you a robust interface and we can write unit tests for this function later:

```python
def artist_exists(self):
    """Checks if an artist exists.

    Returns:
        True if the artist exists, False otherwise.
    """

    conn = sqlite3.connect(DB_NAME)
    curs = conn.cursor()
    
    curs.execute("select * from Artists where a_name=?", (self.a_name, ))
    artist = curs.fetchone()
    
    curs.close()
    conn.close()

    return artist is not None
```

Reply:
fixed